### PR TITLE
doc(bdk): Clarify the absolute_fee docs

### LINKED ISSUE #1066: Absolute fee constraint in TxBuilder not being respected
When creating a tx the absolute fee amount when set does not always match the actual fee amount. I see this is because the amount is added to several times, so it must be deliberate, but for someone using the library it is unintuitive. What is the rationale?

### PR DESCRIPTION
**Pull request by danielabrozzoni**
Fixes #1066

#### All Submissions:

* [x] I've signed all my commits
* [x] I followed the [contribution guidelines](https://github.com/bitcoindevkit/bdk/blob/master/CONTRIBUTING.md)
* [x] I ran `cargo fmt` and `cargo clippy` before committing

### CODE REVIEW

File: crates/bdk/src/wallet/tx_builder.rs [hunk 1]
```diff
@@ -198,6 +198,10 @@ impl<'a, D, Cs: CoinSelectionAlgorithm, Ctx: TxBuilderContext> TxBuilder<'a, D,
     /// If anyone sets both the fee_absolute method and the fee_rate method,
     /// the FeePolicy enum will be set by whichever method was called last,
     /// as the FeeRate and FeeAmount are mutually exclusive.
+    ///
+    /// None that this is a really a minimum absolute fee -- it's possible to
```
Comment (thunderbiscuit) [inline crates/bdk/src/wallet/tx_builder.rs]: I think this is a typo: `None => Note`

Comment (danielabrozzoni) [reply crates/bdk/src/wallet/tx_builder.rs]: Ooops, thanks for spotting it!

File: crates/bdk/src/wallet/tx_builder.rs [hunk 2]
```diff
@@ -198,6 +198,10 @@ impl<'a, D, Cs: CoinSelectionAlgorithm, Ctx: TxBuilderContext> TxBuilder<'a, D,
     /// If anyone sets both the fee_absolute method and the fee_rate method,
     /// the FeePolicy enum will be set by whichever method was called last,
     /// as the FeeRate and FeeAmount are mutually exclusive.
+    ///
+    /// Note that this is a really a minimum absolute fee -- it's possible to
```
Comment (kaiwolfram) [inline crates/bdk/src/wallet/tx_builder.rs]:
```suggestion
    /// Note that this is really a minimum absolute fee -- it's possible to
```

Comment (danielabrozzoni) [reply crates/bdk/src/wallet/tx_builder.rs]: Ah! Turns out I had two different typos in the same line 😅  Fixed, thanks!

File: crates/bdk/src/wallet/tx_builder.rs [hunk 3]
```diff
@@ -198,6 +198,10 @@ impl<'a, D, Cs: CoinSelectionAlgorithm, Ctx: TxBuilderContext> TxBuilder<'a, D,
     /// If anyone sets both the fee_absolute method and the fee_rate method,
     /// the FeePolicy enum will be set by whichever method was called last,
     /// as the FeeRate and FeeAmount are mutually exclusive.
+    ///
+    /// Note that this is really a minimum absolute fee -- it's possible to
+    /// overshoot it slightly since adding a change output to drain the remaining
+    /// excess might not be viable.
```
Comment (evanlinjin) [inline crates/bdk/src/wallet/tx_builder.rs:208]: Couldn't we say this about the feerate as well?

Comment (danielabrozzoni) [reply crates/bdk/src/wallet/tx_builder.rs:208]: Fair, I updated the `fee_rate` description, lmk if it makes sense to you

**Review by notmandatory — APPROVED** [review]
ACK 0ecc0280c032301dc638e75e54a8691d66d50274

New docs makes sense to me.

**Pull request merged**